homeassistant/components/connector/cover.py

```diff
@@ -0,0 +1,293 @@
+            entities.append(
+                TwoWayDevice(
+                    coordinator=coordinator,
+                    blind=hub,
+                    device_class=wifi_motor_type,
+                    config_entry=config_entry,
+                )
+            )
+        else:
+            for blind in hub.blinds_list.values():
+                blind_type = POSITION_DEVICE_MAP.get(blind.type, CoverDeviceClass.SHADE)
+                if blind.wireless_mode in ONEWAYWIRELESSMODE:
+                    entities.append(
+                        OneWayDevice(
+                            coordinator=coordinator,
+                            blind=blind,
+                            device_class=blind_type,
+                            config_entry=config_entry,
+                        )
+                    )
+                elif blind.wireless_mode in TWOWAYWIRELESSMODE:
+                    if blind.type in VENETIANTYPE:
+                        entities.append(
+                            TiltDevice(
+                                coordinator=coordinator,
+                                blind=blind,
+                                device_class=blind_type,
+                                config_entry=config_entry,
+                            )
+                        )
+                    else:
+                        entities.append(
+                            TwoWayDevice(
+                                coordinator=coordinator,
+                                blind=blind,
+                                device_class=blind_type,
+                                config_entry=config_entry,
+                            )
+                        )
+                else:
+                    _LOGGER.info("This wirelessMode not support")
+
+    async_add_entities(entities)
+
+
+class TwoWayDevice(CoordinatorEntity, CoverEntity):
+    """Representation of a Motion Blind Device."""
+
+    def __init__(self, coordinator, blind, device_class, config_entry):
+        """Initialize the blind."""
+        super().__init__(coordinator)
+
+        self._blind = blind
+        self._device_class = device_class
+        self._config_entry = config_entry
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the blind."""
+        return self._blind.mac
+
+    @property
+    def device_info(self):
+        """Return the device info of the blind."""
+        device_info = {
+            "identifiers": {(DOMAIN, self._blind.mac)},
+            "manufacturer": MANUFACTURER,
+            "name": f"{self._blind.mac}",
+            "model": "Curtain",
+            "via_device": (DOMAIN, self._config_entry.unique_id),
+        }
+
+        return device_info
+
+    @property
+    def name(self):
+        """Return the name of the blind."""
+        return f"{self._blind.mac[-4:]}"
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return True
```

Comment:
Maybe you should check the connection state in some global way?

Reply:
Because the Connector integration obtains the motor list through the HUB local UDP interface (the HUB and the motor communicate through 433MHZ), it is impossible to judge whether the motor is available through this interface, so by default, as long as the motor list is obtained, it means that the motor is available